File: legacy/006.c
// 25164150
#include <stdio.h>

#define square(x) (x)*(x)

int main()
{
    const int n = 100;
    int square_sum = square(n*(n + 1)/2);
    int sum_square = n*(n + 1)*(2*n + 1)/6;
    printf("%d\n", square_sum - sum_square);
    return 0;
}

File: zig/006.zig
const std = @import("std");
const stdout = std.io.getStdOut().writer();

fn sumSquare(n: u64) u64 {
    const sum = (n * n * n / 3) + (n * n / 2) + (n / 6);
    return sum;
}

fn squareSum(n: u64) u64 {
    const sum = n * (n + 1) / 2;
    return sum * sum;
}

fn solve(n: u64) u64 {
    return squareSum(n) - sumSquare(n) - 1;
}

pub fn main() !void {
    try stdout.print("{}\n", .{solve(100)});
}
